### DISCUSSION TITLE
How can we implement session stickiness for kong 1.1.1 version

### DISCUSSION THREAD

**Discussion by aryagaurav001** [Kong/kong#10930] [Help]
Can some one please guide if the session stickiness can be implemented in Kong 1.1.1 version. I understand that the docs have been archived and is unavailable. The algorithm attribute does not exist on upstream. I am referring the fork 1.1.1 version which supports mariadb as the database.

curl -X POST http://localhost:8001/upstreams -d name=example_upstream -d algorithm=consistent-hashing
{"message":"schema violation (algorithm: unknown field)","name":"schema violation","fields":{"algorithm":"unknown field"},"code":2}

**Comment by Tieske** [comment]
check the source code at that version; https://github.com/Kong/kong/blob/1.1.1/kong/db/schema/entities/upstreams.lua#L149

the algorithm field didn't exist. So you'd need to set `hash_on` ; https://github.com/Kong/kong/blob/1.1.1/kong/db/schema/entities/upstreams.lua#L24
as well as the appropriate target (in case of a header or cookie)
